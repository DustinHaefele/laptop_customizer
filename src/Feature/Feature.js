import React from 'react';

export default function Feature(props) {

  let options = props.options.map((item, index) => 
  {
    const selectedClass =
      item.name === props.selected[props.key].name
        ? "feature__selected"
        : "";
    const featureClass = "feature__option " + selectedClass;
    return (
      <li key={index} className="feature__item">
        <div
          className={featureClass}
          onClick={(e) => props.handleUpdate(props.key, item)}
        >
          {item.name}(
          {new Intl.NumberFormat("en-US", {
            style: "currency",
            currency: "USD"
          }).format(item.cost)}
          )
        </div>
      </li>
    );
  });

  return (
    <div className="feature" key={props.key}>
      <div className="feature__name">{props.key}</div>
      <ul className="feature__list">{options}</ul>
    </div>
  );
}

