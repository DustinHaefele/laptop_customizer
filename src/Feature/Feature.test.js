import React from "react";
import ReactDOM from "react-dom";
import Enzyme from "enzyme";
import toJson from "enzyme-to-json";
import Feature from "./Feature";

describe("Feature Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Feature
        options={[
          {
            name: "17th Generation Intel Core HB (7 Core with donut spare)",
            cost: 700
          },
          {
            name: "Professor X AMD Fire Breather with sidewinder technology",
            cost: 1200
          }
        ]}
        myKey={"Operating System"}
        selected={{
          Processor: {
            name: "17th Generation Intel Core HB (7 Core with donut spare)",
            cost: 700
          },
          "Operating System": {
            name: "Ubuntu Linux 16.04",
            cost: 200
          },
          "Video Card": {
            name: "Toyota Corolla 1.5v",
            cost: 1150.98
          },
          Display: {
            name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
            cost: 1500
          }
        }}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders UI as expected", () => {
    const wrapper = Enzyme.shallow(
      <Feature
        options={[
          {
            name: "17th Generation Intel Core HB (7 Core with donut spare)",
            cost: 700
          },
          {
            name: "Professor X AMD Fire Breather with sidewinder technology",
            cost: 1200
          }
        ]}
        myKey={"Processor"}
        selected={{
          Processor: {
            name: "17th Generation Intel Core HB (7 Core with donut spare)",
            cost: 700
          },
          "Operating System": {
            name: "Ubuntu Linux 16.04",
            cost: 200
          },
          "Video Card": {
            name: "Toyota Corolla 1.5v",
            cost: 1150.98
          },
          Display: {
            name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
            cost: 1500
          }
        }}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Changes which is selected onClick", () => {
    const wrapper = Enzyme.shallow(
      <Feature
        options={[
          {
            name: "17th Generation Intel Core HB (7 Core with donut spare)",
            cost: 700
          },
          {
            name: "Professor X AMD Fire Breather with sidewinder technology",
            cost: 1200
          }
        ]}
        myKey={"Processor"}
        selected={{
          Processor: {
            name: "17th Generation Intel Core HB (7 Core with donut spare)",
            cost: 700
          },
          "Operating System": {
            name: "Ubuntu Linux 16.04",
            cost: 200
          },
          "Video Card": {
            name: "Toyota Corolla 1.5v",
            cost: 1150.98
          },
          Display: {
            name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
            cost: 1500
          }
        }}
        onClick={jest.fn()}
      />
    );
    wrapper
      .find("div")
      .at(1)
      .simulate("click");
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders with the correct selected class", () => {
    const wrapper = Enzyme.shallow(
      <Feature
        options={[
          {
            name: "17th Generation Intel Core HB (7 Core with donut spare)",
            cost: 700
          },
          {
            name: "Professor X AMD Fire Breather with sidewinder technology",
            cost: 1200
          }
        ]}
        myKey={"Processor"}
        selected={{
          Processor: {
            name: "Professor X AMD Fire Breather with sidewinder technology",
            cost: 1200
          },
          "Operating System": {
            name: "Ubuntu Linux 16.04",
            cost: 200
          },
          "Video Card": {
            name: "Toyota Corolla 1.5v",
            cost: 1150.98
          },
          Display: {
            name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
            cost: 1500
          }
        }}
        onClick={jest.fn()}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
