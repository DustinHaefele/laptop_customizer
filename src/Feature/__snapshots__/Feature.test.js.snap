// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature Component Changes which is selected onClick 1`] = `
<div
  className="feature"
>
  <div
    className="feature__name"
  />
  <ul
    className="feature__list"
  >
    <li
      className="feature__item"
      key="0"
    >
      <div
        className="feature__option feature__selected"
        onClick={[Function]}
      >
        17th Generation Intel Core HB (7 Core with donut spare)
        (
        $700.00
        )
      </div>
    </li>
    <li
      className="feature__item"
      key="1"
    >
      <div
        className="feature__option "
        onClick={[Function]}
      >
        Professor X AMD Fire Breather with sidewinder technology
        (
        $1,200.00
        )
      </div>
    </li>
  </ul>
</div>
`;

exports[`Feature Component renders UI as expected 1`] = `
<div
  className="feature"
>
  <div
    className="feature__name"
  />
  <ul
    className="feature__list"
  >
    <li
      className="feature__item"
      key="0"
    >
      <div
        className="feature__option feature__selected"
        onClick={[Function]}
      >
        17th Generation Intel Core HB (7 Core with donut spare)
        (
        $700.00
        )
      </div>
    </li>
    <li
      className="feature__item"
      key="1"
    >
      <div
        className="feature__option "
        onClick={[Function]}
      >
        Professor X AMD Fire Breather with sidewinder technology
        (
        $1,200.00
        )
      </div>
    </li>
  </ul>
</div>
`;

exports[`Feature Component renders with the correct selected class 1`] = `
<div
  className="feature"
>
  <div
    className="feature__name"
  />
  <ul
    className="feature__list"
  >
    <li
      className="feature__item"
      key="0"
    >
      <div
        className="feature__option "
        onClick={[Function]}
      >
        17th Generation Intel Core HB (7 Core with donut spare)
        (
        $700.00
        )
      </div>
    </li>
    <li
      className="feature__item"
      key="1"
    >
      <div
        className="feature__option feature__selected"
        onClick={[Function]}
      >
        Professor X AMD Fire Breather with sidewinder technology
        (
        $1,200.00
        )
      </div>
    </li>
  </ul>
</div>
`;
